source/common/protobuf/utility.cc

```diff
@@ -366,34 +366,6 @@ class UnexpectedFieldProtoVisitor : public ProtobufMessage::ConstProtoVisitor {
         (!field.is_repeated() && !reflection->HasField(message, &field))) {
       return nullptr;
     }
-
-    // If this field is deprecated, warn or throw an error.
-    if (field.options().deprecated()) {
-      if (absl::StartsWith(field.name(), "hidden_envoy_deprecated_")) {
-        // The field was marked as hidden_envoy_deprecated and an error must be thrown,
-        // unless it is part of an explicit test that needs access to the deprecated field
-        // when we enable runtime deprecation override to allow point field overrides for tests.
-        if (!runtime_ ||
-            !runtime_->snapshot().deprecatedFeatureEnabled(
-                absl::StrCat("envoy.deprecated_features:", field.full_name()), false)) {
-          const std::string fatal_error = absl::StrCat(
-              "Illegal use of hidden_envoy_deprecated_ V2 field '", field.full_name(),
-              "' from file ", filename,
-              " while using the latest V3 configuration. This field has been removed from the "
-              "current Envoy API. Please see " ENVOY_DOC_URL_VERSION_HISTORY " for details.");
-          throw ProtoValidationException(fatal_error, message);
-        }
-      }
-      const std::string warning =
```

Comment:
@tyxia you can delete the above `if` condition, but we should still retain deprecated field handling here, which is not connected with major versioning and will also be useful with minor versioning.